.github/workflows/src/verify-run-status.js

```diff
@@ -0,0 +1,181 @@
+import { extractInputs } from "./context.js";
+import { PER_PAGE_MAX } from "./github.js";
+
+/**
+ * @typedef {import('@octokit/plugin-rest-endpoint-methods').RestEndpointMethodTypes} RestEndpointMethodTypes
+ * 
+ * @typedef {RestEndpointMethodTypes["checks"]["listForRef"]["parameters"]} ListCheckRunsForRefParameters
+ * @typedef {RestEndpointMethodTypes["checks"]["listForRef"]["response"]["data"]["check_runs"][number]} CheckRun
+
+ * @typedef {RestEndpointMethodTypes["actions"]["listWorkflowRunsForRepo"]["parameters"]} ListWorkflowRunsForRepoParameters
+ * @typedef {RestEndpointMethodTypes["actions"]["listWorkflowRunsForRepo"]["response"]["data"]["workflow_runs"][number]} WorkflowRun
+ */
+
+/**
+ *
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ */
+export async function verifyRunStatus({ github, context, core }) {
+  const checkRunName = process.env.CHECK_RUN_NAME || "";
```

Comment:
```suggestion
  const checkRunName = process.env.CHECK_RUN_NAME;
```

If you're going to throw if not set, I don't think you need a default value.